language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ClientResourceAccessActionNodeContext.java

```diff
@@ -211,7 +211,27 @@ public void sort(BallerinaCompletionContext context, ClientResourceAccessActionN
             return;
         }
 
-        super.sort(context, node, completionItems);
+        Optional<TypeSymbol> parameterSymbol = getParameterTypeSymbol(context);
+        int rankOffset = 1;
+        for (LSCompletionItem completionItem : completionItems) {
+            if (completionItem.getType() == LSCompletionItem.CompletionItemType.NAMED_ARG) {
+                sortNamedArgCompletionItem(context, completionItem);
+            } else if (parameterSymbol.isEmpty()) {
+                sortParameterlessCompletionItem(context, completionItem);
+            } else if (completionItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
+                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) completionItem;
+                if (symbolCompletionItem.getSymbol().isPresent() &&
+                        symbolCompletionItem.getSymbol().get().kind() == SymbolKind.RESOURCE_METHOD) {
+                    completionItem.getCompletionItem().setSortText(
+                            SortingUtil.genSortTextByAssignability(context, completionItem, parameterSymbol.get()) +
+                                    SortingUtil.genSortText(rankOffset));
+                }
+                sortDefaultCompletionItem(context, parameterSymbol.get(), completionItem);
+                rankOffset++;
```

Comment:
If it is not mandatory to increase by one, we can do https://github.com/ballerina-platform/ballerina-lang/pull/41663/files#r1402992223 here as well.